### PR TITLE
Add protections against infinite loop in bezier calculations

### LINKED ISSUE #27753: [Bug]:  `split_bezier_intersecting_with_closedpath` fails to return
### Bug summary

It looks like, for some inputs, `split_bezier_intersecting_with_closedpath`  never returns, as the `while True:` loop inside `find_bezier_t_intersecting_with_closedpath` fails to converge.

### Code for reproduction

```python
from matplotlib.patches import inside_circle, split_bezier_intersecting_with_closedpath

arrow_path = [(96950809781500.0, 804.7503795623779), (96950809781500.0, 859.6242585800646), (96950809781500.0, 914.4981375977513)]
in_f = inside_circle(96950809781500.0, 804.7503795623779, 0.06)
split_bezier_intersecting_with_closedpath(arrow_path, in_f)
```

### Actual outcome

The call to `split_bezier_intersecting_with_closedpath` never returns.

### Expected outcome

The call to `split_bezier_intersecting_with_closedpath` should return the split segment.

### Additional information

_No response_

### Operating system

confirmed on Arch and Ubuntu 22.04 LTS

### Matplotlib Version

3.8.2

### Matplotlib Backend

agg

### Python version

3.11.6 (Arch) and 3.10.12 (Ubuntu)

### Jupyter version

4.0.11  (note: reproducible outside jupyter)

### Installation

None

### PR DESCRIPTION
**Pull request by ksunden**
Closes #27753

## PR summary

This is usually an internal detail of things like `FancyArrowPatch`. The failure mode
is an infinite loop. Since the only case we are changing never worked, no API note is
included

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [x] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [x] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Review by tacaswell — APPROVED** [review]

**Comment by ksunden** [comment]
I would like to push forward on the release of 3.8.3, can this get a second review, please?

**Review by greglucas — APPROVED** [review]

**Pull request merged**
